File: julia/search_linear.jl
function linearSearch(givenSequnce::String, searchTerm::String)
    #= Linear Search Function. =#
    lengthOfSearchTem = length(searchTerm)
    lenghtOfGivenSequnce = length(givenSequnce)
    leftCursor = 1
    rightCursor = lengthOfSearchTem
    while rightCursor <= lenghtOfGivenSequnce
        if SubString(givenSequnce, leftCursor, rightCursor - 1) == searchTerm
            return true
        end
        leftCursor += 1
        rightCursor = leftCursor + lengthOfSearchTem
    end
    return false
end

function main()
    #= Main. =#
    SAMPLE_INPUT_SEQUENCE = "This is an example sequence."
    SAMPLE_INPUT_SEARCH_TERM = "an ex"

    print("Sequnce: ")
    inputSequence = strip(readline())
    print("Search term: ")
    inputSearchTerm = strip(readline())

    if inputSequence == ""
        inputSequence = SAMPLE_INPUT_SEQUENCE
    end
    if inputSearchTerm == ""
        inputSearchTerm = SAMPLE_INPUT_SEARCH_TERM
    end

    println(linearSearch(inputSequence, inputSearchTerm))

end

main()

File: dart/search_linear.dart
import 'dart:convert';
import "dart:io";

bool linearSearch(String givenSequence, String searchTerm) {
  /* Linear Search Function. */
  final lengthOfSearchTerm = searchTerm.length;
  final lengthOfGivenSequence = givenSequence.length;
  var leftCursor = 0;
  var rightCursor = lengthOfSearchTerm;

  while (rightCursor <= lengthOfGivenSequence) {
    if (givenSequence.substring(leftCursor, rightCursor) == searchTerm) {
      return true;
    }
    leftCursor += 1;
    rightCursor = leftCursor + lengthOfSearchTerm;
  }
  return false;
}

void main() {
  /* Main. */
  const SAMPLE_INPUT_SEQUENCE = "This is an example sequence.";
  const SAMPLE_INPUT_SEARCH_TERM = "an ex";

  stdout.write("Sequence: ");
  var inputSequence = (stdin.readLineSync(encoding: utf8))?.trim();
  stdout.write("Search term: ");
  var inputSearchTerm = (stdin.readLineSync(encoding: utf8))?.trim();

  if (inputSequence == null || inputSequence.length == 0) {
    inputSequence = SAMPLE_INPUT_SEQUENCE;
  }
  if (inputSearchTerm == null || inputSearchTerm.length == 0) {
    inputSearchTerm = SAMPLE_INPUT_SEARCH_TERM;
  }

  print(linearSearch(inputSequence, inputSearchTerm));
}
